### DISCUSSION TITLE
Localization in French

### DISCUSSION THREAD

**Discussion by Lidgeu** [GameGodS3/DropPoint#22] [Q&A]
Hello,
I just discovered today your powerful and very useful DropPoint (1.1.3) portable utility.
I'm one of the administrator of the number one French freeware translations websit - free of charge :o) - : [**Colok Traductions**](url=https://www.colok-traductions.com/). And, despite the very few English messages to translate, the "typical" French users are very reluctant to (ignorant of ?) foreign languages :o)).
In your zip, there is a "locales" folder, but the French one content looks like nothing to do with DropPoint and anyway how to localize is not documented.
My question : how can we have/build a French version of DropPoint. If possible, I'm sure it will interest our members.

Bets regards and thanks beforehand for your reply (whatever it will be)

Pierre

**Comment by GameGodS3** [comment]
Wow, I would **love** to include localisation in DropPoint although I haven't thought of prioritizing it further up. Since you've made a request like so, I think it's important that we work on making translated UI as quickly as possible.
And no, I don't find that as being "ignorant" per se. It's only my pleasure to have DropPoint in multiple languages :) . 
However, I haven't yet looked into how localisation can be implemented in ElectronJS. So far I have found [this article](https://phrase.com/blog/posts/building-an-electron-app-with-internationalization-i18n/) which implements something similar. Only roadblock here is, I don't know how will I do the translation inside plain html/css. I am not using React. I think we can figure it out together.

If we could collaborate and look into it further, I wouldn't hesitate to add a French version of DropPoint. Thank you for the suggestion.

Sudev

**Comment by Lidgeu** [comment]
Hi Sudev,
To be clear : I am not a programmer at all (despite I worked for a multinational I.T. Company for decades, before my retirement).

However I link to article you mentioned. ElectronJS looks like powerful but I'm not sure such a tool is necessary for the couple of messages DropPoint is using.
Of course, going to multi-localisation of an existing software is not "trivial". The developer has to deal with languages particularities (f.i. in average, French (German) needs more width than English for the same meaning --> the fields width to allocate are larger, to avoid truncation), singular/plural management (see the example below), character set(s), etc. 
Nevertheless, numerous developers use the "individual language file approach" where the user got the option to choose "other than English by default" and dynamically load the selected language file. 
Here is an example of what could be such language files : [DropPoint_1.1.3-FR.zip](https://github.com/GameGodS3/DropPoint/files/7907325/DropPoint_1.1.3-FR.zip)
FYI : the very simple xml layout is the one used by Gaku (Tablacus Explorer - also on Github).
To conclude : 
> I unfortunately have no abilities to contribute to the coding, BUT
> I'm open to play the role of "early tester" of a French localisation implementation. 
;o)
Pierre
